### PR TITLE
fix typo

### PR DESCRIPTION
**Pull request by fdopen**
I think, it's a simple typo. UV_HANDLE_CONNECTED is defined, but never used anywhere. So this test can't be true. UV_HANDLE_CONNECTION was meant.

You can find an example, that doesn't work because of it here: https://gist.github.com/fdopen/642121f59782d5b668a6

### CODE REVIEW

**Comment by saghul** [comment]
I'll investigate. If this was a typo, then that branch would never be executed.

**Comment by saghul** [comment]
Turns out this "typo" goes at least all the way back to libuv 0.6 days. Also, that branch would only get executed when UV_INHERIT_STREAM is used, which is on one of the libuv tests :-( One should be added in order to validate this fix.

**Comment by fdopen** [comment]
Ok, I've added a test for it.

What is/was the rational behind `UV_HANDLE_CONNECTED`? Can't it be removed entirely?

**Comment by saghul** [comment]
> Ok, I've added a test for it.
> 

Great, thanks! I'll have a look as soon as I can.

> What is/was the rational behind UV_HANDLE_CONNECTED? Can't it be removed entirely?

TBH I don't know. It can be removed, since it's not in use.

**Comment by saghul** [comment]
CI run: https://jenkins-iojs.nodesource.com/view/libuv/job/libuv+any-pr+multi/112/

**Comment by saghul** [comment]
Test is passing :+1:

File: test/test-spawn.c [hunk 1]
```diff
@@ -1540,3 +1540,168 @@ TEST_IMPL(spawn_reads_child_path) {
   MAKE_VALGRIND_HAPPY();
   return 0;
 }
+
+#ifndef _WIN32
+static int mpipe(int *fds)
+{
+  if ( pipe(fds) == -1 ){
```
Comment (saghul) [inline test/test-spawn.c]: Remove the space between the parenthesis and "pipe". There are more occurrences of this.

File: test/test-spawn.c [hunk 2]
```diff
@@ -1540,3 +1540,168 @@ TEST_IMPL(spawn_reads_child_path) {
   MAKE_VALGRIND_HAPPY();
   return 0;
 }
+
+#ifndef _WIN32
+static int mpipe(int *fds)
+{
```
Comment (saghul) [inline test/test-spawn.c]: Put the brace on the same line.

File: test/test-spawn.c [hunk 3]
```diff
@@ -1540,3 +1540,168 @@ TEST_IMPL(spawn_reads_child_path) {
...
+  return 0;
+}
+
+/* Helper for child process of spawn_inherit_streams */
+#ifndef _WIN32
+int spawn_stdin_stdout(void)
+{
+  char buf[1024];
+  char *pbuf;
+  for (;;){
```
Comment (saghul) [inline test/test-spawn.c]: Space between closing parenthesis and open brace.

File: test/test-spawn.c [hunk 4]
```diff
@@ -1540,3 +1540,168 @@ TEST_IMPL(spawn_reads_child_path) {
...
+    ssize_t r,w,c;
+    do {
+      r = read(0,buf,sizeof buf);
```
Comment (saghul) [inline test/test-spawn.c]: Space after comma.

**Comment by saghul** [comment]
Change looks solid, thanks a lot for writing the test! Can you please take care of the style issues?

**Comment by fdopen** [comment]
Ok,
I've removed  `UV_HANDLE_CONNECTED` and changed the style.

**Comment by saghul** [comment]
Thanks @fdopen, landed in 15e4ded!

**Pull request closed without merging**